XamarinCommunityToolkit/Helpers/WeakEventManager.shared.cs

```diff
@@ -0,0 +1,152 @@
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+
+using static System.String;
+
+namespace Xamarin.CommunityToolkit.Helpers
+{
+	// Copy from https://github.com/xamarin/Xamarin.Forms/blob/5.0.0/Xamarin.Forms.Core/WeakEventManager.cs
+	// TODO: Remove when PR complete: https://github.com/xamarin/Xamarin.Forms/pull/12063
+	class WeakEventManager
+	{
+		readonly Dictionary<string, List<Subscription>> eventHandlers = new Dictionary<string, List<Subscription>>();
+
+		public void AddEventHandler<TEventArgs>(EventHandler<TEventArgs> handler,
+			[CallerMemberName] string eventName = null)
+			where TEventArgs : EventArgs
+		{
+			if (IsNullOrEmpty(eventName))
+				throw new ArgumentNullException(nameof(eventName));
```

Comment:
For the string checks, can we just return instead of throwing an exception? (This apply to the others)